api/osf_api.py

```diff
@@ -1050,3 +1050,136 @@ def get_project_node_analytics_data(session, node_id=None, timespan='week'):
     url = '_/metrics/query/node_analytics/{}/{}/'.format(node_id, timespan)
     data = session.get(url)['data']
     return data or None
+
+
+def get_fake_file_guid(session, file_id):
+
+    url = '/v2/files' + file_id
+    data = session.get(url=url)
+    file_guid = data['data']['attributes']['guid']
+    return file_guid
+
+
+def get_existing_file_data(session, node_id=settings.PREFERRED_NODE):
+    """Return the id of the first file in OSFStorage on a given node.
+    Uploads a new file if one does not exist.
+    """
+    node = client.Node(session=session, id=node_id)
+    node.get()
+    files_url = node.relationships.files['links']['related']['href']
+    data = session.get(files_url + 'osfstorage/')
+    name = data['data'][0]['attributes']['name']
+    return name, data
+
+
+def update_custom_project_metadata(session, node_id):
+    """Updates project metadata fields resource_type and
+    resource_language and support funder info with custom values"""
+    url = 'v2/custom_item_metadata_records/{}/'.format(node_id)
+    raw_payload = {
+        'data': {
+            'id': node_id,
+            'type': 'custom-item-metadata-records',
+            'attributes': {
+                'language': 'eng',
+                'resource_type_general': 'Collection',
+            },
+        }
+    }
+    session.put(url=url, raw_body=json.dumps(raw_payload))
+
+
+def delete_project_contributor(session, node_id, user_name):
+    """This method deletes the given contributor
+    from contributors list for the given project guid"""
+
+    url = '/v2/nodes/{}/contributors/'.format(node_id)
+    data = session.get(url)['data']
+
+    for i in range(0, len(data)):
+        if data[i]['embeds']['users']['data']['attributes']['full_name'] == user_name:
+            user_id = data[i]['embeds']['users']['data']['id']
+            delete_url = '/v2/nodes/{}/contributors/{}/'.format(node_id, user_id)
+            session.delete(delete_url, item_type='users')
+            break
+
+
+def get_most_recent_registration_node_id_by_user(user_name, session):
+    """Return the most recently approved public registration node id by the given user.
+    The /v2/registrations endpoint currently returns the most recently modified
+    registration sorted first. But we still need to check for a public and
+    approved registration that has not been withdrawn in order to get a
+    registration that is fully accessible.
+    """
+    if not session:
+        session = get_default_session()
+    url = '/v2/registrations/?embed=contributors'
+    page_data = session.get(url)['links']
+    total_registrations = page_data['meta']['total']
+    per_page = 10
+    total_pages = round(total_registrations / per_page)
+    for page in range(1, total_pages):
+        page_url = 'v2/registrations/?embed=contributors&page=%s' % page
```

Comment:
let's try a search for title

Reply:
Selenium Registration for Metadata tests

Reply:
Done. Updated get_most_recent_registration_node_id_by_user to search with title (filtering with title in the url)